graph/ui-project-details/src/lib/project-info-header/project-info-header.tsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @nx/enforce-module-boundaries */
+// nx-ignore-next-line
+import type { ProjectGraphProjectNode } from '@nx/devkit';
+import { TagList } from '../tag-list/tag-list';
+import { EyeIcon } from '@heroicons/react/24/outline';
+import { twMerge } from 'tailwind-merge';
+// nx-ignore-next-line
+
+export type ProjectInfoHeaderProps = {
+  project: ProjectGraphProjectNode;
+  onViewInProjectGraph?: (data: { projectName: string }) => void;
+  viewInProjectGraphPosition?: 'top' | 'bottom';
+  isCompact?: boolean;
+};
+
+const typeToProjectType = {
+  app: 'Application',
+  lib: 'Library',
+  e2e: 'E2E',
+};
+
+export function ProjectInfoHeader({
+  project,
+  isCompact,
+  onViewInProjectGraph,
+  viewInProjectGraphPosition,
+}: ProjectInfoHeaderProps) {
+  const projectData = project.data;
+  return (
+    <div
+      className={twMerge(
+        `border-b border-slate-900/10 dark:border-slate-300/10`,
+        isCompact ? 'mb-2' : 'mb-4'
+      )}
+    >
+      <div className="flex flex-wrap justify-between py-2">
+        <div className="min-w-0">
+          {projectData.metadata?.description ? (
+            <p className="mb-2 text-sm capitalize text-gray-500 dark:text-slate-400">
+              {projectData.metadata?.description}
+            </p>
+          ) : null}
+          {projectData.tags && projectData.tags.length ? (
+            <TagList tags={projectData.tags} />
+          ) : null}
+          {projectData.root ? (
+            <p>
+              <span className="font-medium">Root:</span>
+              <span className="font-mono"> {projectData.root.trim()}</span>
+            </p>
+          ) : null}
+          {projectData.projectType ?? typeToProjectType[project.type] ? (
+            <p>
+              <span className="font-medium">Type:</span>
+              <span className="ml-2 font-mono capitalize">
+                {projectData.projectType ?? typeToProjectType[project.type]}
+              </span>
+            </p>
+          ) : null}
+        </div>
+        <div className="self-end">
+          {onViewInProjectGraph && viewInProjectGraphPosition === 'bottom' && (
+            <ViewInProjectGraphButton
+              callback={() =>
+                onViewInProjectGraph({ projectName: project.name })
+              }
+            />
+          )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export function ViewInProjectGraphButton({
+  callback,
+}: {
+  callback: () => void;
```

Comment:
This should be `onClick` as it is a click event handler. Also, I don't  think it's large/complex enough to warrant its own component. Just inline in the main component function.